[AMDGPU] Fix opcode comparison logic for G_INTRINSIC

pbalcer opened the pull request:
The check `(Opc < TargetOpcode::GENERIC_OP_END)` incorrectly
includes `G_INTRINSIC` (129), which is less than
`GENERIC_OP_END` (313), leading to logically dead code.

This patch reorders the conditionals to first check for `G_INTRINSIC`, ensuring
correct handling of the `amdgcn_fdot2` intrinsic.

arsenm commented on llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp:
```diff
@@ -4718,14 +4718,14 @@ class SearchOptions {
       // Only float point intrinsic has neg & neg_hi bits.
       if (IntrinsicID == Intrinsic::amdgcn_fdot2)
         HasNeg = true;
+    } else if (Opc < TargetOpcode::GENERIC_OP_END) {
```
Should probably use TargetInstrInfo::isGenericOpcode? 

If G_INTRINSIC doesn't count as isGenericOpcode, should fix that

pbalcer replied:
Thanks, done.

pbalcer committed:
[AMDGPU] Fix opcode comparison logic for G_INTRINSIC

The check `(Opc < TargetOpcode::GENERIC_OP_END)` incorrectly
includes `G_INTRINSIC` (129), which is less than
`GENERIC_OP_END` (313), leading to logically dead code.

This patch reorders the conditionals to first check for `G_INTRINSIC`, ensuring
correct handling of the `amdgcn_fdot2` intrinsic.

pbalcer committed:
Merge branch 'main' into fix-amdgcn-fdot2
